dbt_platform_helper/utils/platform_config_schema.py

```diff
@@ -0,0 +1,603 @@
+import ipaddress
+import re
+
+from schema import Optional
+from schema import Or
+from schema import Regex
+from schema import Schema
+from schema import SchemaError
+
+
+def _string_matching_regex(regex_pattern: str):
+    def validator(string):
+        if not re.match(regex_pattern, string):
+            # Todo: Raise suitable PlatformException?
+            raise SchemaError(
+                f"String '{string}' does not match the required pattern '{regex_pattern}'. For more details on valid string patterns see: https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/"
+            )
+        return string
+
+    return validator
+
+
+def _integer_between(lower, upper):
+    def _is_between(value):
+        if isinstance(value, int) and lower <= value <= upper:
+            return True
+        # Todo: Raise suitable PlatformException?
+        raise SchemaError(f"should be an integer between {lower} and {upper}")
+
+    return _is_between
```

Comment:
I wonder if these validation functions would be easier to understand at first pass if they always returned something called `validation_function` or something like that?  I found the idea of `_is_between(value)` a bit confusing.

Reply:
Reworked in person.